messages/README.md

```diff
@@ -0,0 +1,84 @@
+# messages
+The `messages` crate provides data structures representing the various messages used in Aries protocols. This `README` explores the architecture behind the crate and module organization, making implementing new messages/protocols easier.
```

Comment:
> ... making implementing new messages/protocols easier.

Understanding the nuts and bolts of a crate of course makes extending it easier, however I wonder it is (or should be) necessary. Perhaps the easiest way to explain how to extend a crate which is designed to be easily extended (like this one) is a PR doing just that - as long as the process is generally systematic. Of course, this doesn't work when it isn't, and get's outdated when the crate's workings are changed substantially (which is a problem with documentation as well).

This is not meant to be actionable, just thoughts.

Reply:
Idk if a reference PR is a good or a bad thing. I never thought of it, to be honest. @Patrik-Stas @gmulhearn what do you think?

Nevertheless, I added an `Extending the crate` section trying to describe the main concepts/steps behind extending the crate. I don't know whether it helps so feedback is welcome.

I'd do the PR thing but I feel like down the line it will get outdated for some reason as we'll keep forgetting about it. And depending on whether you're just adding some message (like `ProblemReport`) to a protocol, adding a minor/major version or adding an entirely new protocol the implementation will differ a bit. So then you either have multiple PR's for each or one PR where you do all and it might actually end up being a bit confusing as well.

Reply:
Yeah as you both mentioned, there's the problem of getting outdated. I think adding `Extending the crate` is good enough.